homeassistant/components/tractive/__init__.py

```diff
@@ -0,0 +1,155 @@
+"""The tractive integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+
+import aiotractive
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_EMAIL, CONF_PASSWORD, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import (
+    DOMAIN,
+    RECONNECT_INTERVAL,
+    SERVER_AVAILABLE,
+    SERVER_UNAVAILABLE,
+    TRACKER_HARDWARE_STATUS_UPDATED,
+    TRACKER_POSITION_UPDATED,
+)
+
+PLATFORMS = ["device_tracker"]
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up tractive from a config entry."""
+    data = entry.data
+
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    client = aiotractive.Tractive(data[CONF_EMAIL], data[CONF_PASSWORD])
```

Comment:
```suggestion
    client = aiotractive.Tractive(data[CONF_EMAIL], data[CONF_PASSWORD], session=async_get_clientsession(hass))
```

Reply:
Is a recommended way? My library which my other integration [Syncthing](https://github.com/home-assistant/core/pull/38331) is based on also supports passing a session or an event loop to the client, but I was asked to not pass a loop, didn't try with a session.

Reply:
It is recommended (and requires for platinum level: https://developers.home-assistant.io/docs/integration_quality_scale_index/#platinum- )